src/main.c

```diff
@@ -742,20 +742,23 @@ main(int argc, char** argv)
       errx(1, "Failed to start logging");
    }
 
-   if (pgagroal_init_prometheus(&prometheus_shmem_size, &prometheus_shmem))
+   if (config->common.metrics > 0)
```

Comment:
should we export `is_prometheus_active` function and use it so the code is consistent?

Reply:
Yeah can be done, but won't it waste some memory and computation as `config` is already there in the program it will again define `struct configuration` in the check function.

However, as far as consistency is concerned it can be done.

Reply:
> Yeah can be done, but won't it waste some memory and computation as `config` is already there in the program it will again define `struct configuration` in the check function.

The only waste I can see is the function call, but I don't think we are optimizing at this level right now.
It could also be a macro.

Reply:
> The only waste I can see is the function call, but I don't think we are optimizing at this level right now.

Okay :+1: 

> It could also be a macro.

Not sure how a macro could solve the issue, can you please elaborate a little.

Also, if we add `NULL` check of `prometheus` in `is_prometheus_enabled()` function (as per the aforementioned change request) then we can't use it in this scenario as it will always fail the test (since `prometheus` is checked before it is being initialised), and the only test which should be checked is `metrics > 0`. So I believe the current setup is ok.

What do you think about it? @fluca1978.

Reply:
> Also, if we add `NULL` check of `prometheus` in `is_prometheus_enabled()` function (as per the aforementioned change request) then we can't use it in this scenario as it will always fail the test (since `prometheus` is checked before it is being initialised), and the only test which should be checked is `metrics > 0`. So I believe the current setup is ok.

Indeed having the test against nulls will fail in this case, so better leave it off the `is_prometheus_enabled`. Sorry, I was running out of coffee.